lib/elastic_record/relation.rb

```diff
@@ -51,11 +51,20 @@ def to_a
       @records ||= find_hits(search_hits)
     end
 
-    def find_hits(search_hits)
+    def load_safe_hits
+      @records ||= find_hits(search_hits, safe: true)
```

Comment:
This method's name, as well as the context where it would typically be used, give no hint that calling the method has side effects on the object it's called on.  If I saw this method called somewhere, I would think it just returns the safe records, without changing state.

How about instead defining a method that changes the state of the class, without returning any value.  i.e.

```ruby
  def safe!
    @safe = true
  end

  def safe?
    @safe
  end

  def find_hits(search_hits)
    ...
    if safe?
      klass.where(id: ids).to_a
    else
      klass.find(ids)
    end
  end
```

Reply:
Nice! I like this interface a lot better.